### PR TITLE
 [ClangImporter] Allow setup without RuntimeResourcePath in Linux 

### PR DESCRIPTION
**Pull request by rintaro**
This is required to enable  refactoring action `ExtractFunction` in Linux: `RefactoringActionExtractFunction::performChange()` setup `CompilerInstance` without `RuntimeResourcePath` for `performParseOnly()`.

### CODE REVIEW

**Comment by rintaro** [comment]
@jrose-apple What do you think?

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please test

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -536,30 +536,32 @@ getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
...
-      // FIXME: Emit a warning of some kind.
+    if (!searchPathOpts.RuntimeResourcePath.empty()) {
+      // The module map used for Glibc depends on the target we're compiling
+      // for, and is not included in the resource directory with the other
+      // implicit module maps. It's at {freebsd|linux}/{arch}/glibc.modulemap.
+      SmallString<128> GlibcModuleMapPath;
+      GlibcModuleMapPath = searchPathOpts.RuntimeResourcePath;
+
+      // Running without a resource directory is not a supported configuration.
+      assert(!GlibcModuleMapPath.empty());
```
Comment (rintaro) [inline lib/ClangImporter/ClangImporter.cpp]: Forgot to remove this assertion...

**Comment by jrose-apple** [comment]
Is it possible to not initialize the importer at all instead? I suspect the answer is "no, a lot of things implicitly assume it exists", but if it is possible that would be a better answer.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
Uh, hm, that's interesting. I forgot we set up the importer so early. I was hoping it happened in `performSema`, and then we could just not do it in `performParseOnly`. Now I'm unhappy that we're changing the behavior of the compiler based on a flag that really should never be absent.

**Comment by jrose-apple** [comment]
That is, RuntimeResourcePath isn't about Clang; it's about Swift. The rest of the compiler is allowed to fall over without a runtime resource path too. I wouldn't *expect* it to do so for `performParseOnly`, and we could decide to keep it from doing so, but it certainly hasn't been a goal.

**Comment by rintaro** [comment]
Hm, thanks, I will think about it.
Closing for now.

**Comment by jrose-apple** [comment]
Sorry to get in the way of whatever you actually want to do!

**Pull request closed without merging**